lib/PuppeteerSharp.Tests/Input/InputTests.cs

```diff
@@ -0,0 +1,554 @@
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/input/button.html");
+            var exception = await Assert.ThrowsAsync<SelectorException>(()
+                => Page.ClickAsync("button.does-not-exist"));
+            Assert.Equal("No node found for selector: button.does-not-exist", exception.Message);
+            Assert.Equal("button.does-not-exist", exception.Selector);
+        }
+
+        // https://github.com/GoogleChrome/puppeteer/issues/161
+        [Fact]
+        public async Task ShouldNotHangWithTouchEnabledViewports()
+        {
+            await Page.SetViewport(TestConstants.IPhone.ViewPort);
+            await Page.Mouse.DownAsync();
+            await Page.Mouse.MoveAsync(100, 10);
+            await Page.Mouse.UpAsync();
+        }
+
+        [Fact]
+        public async Task ShouldTypeIntoTheTextarea()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/input/textarea.html");
+
+            var textarea = await Page.GetElementAsync("textarea");
+            await textarea.TypeAsync("Type in this text!");
+            Assert.Equal("Type in this text!", await Page.EvaluateExpressionAsync<string>("result"));
+        }
+
+        [Fact]
+        public async Task ShouldClickTheButtonAfterNavigation()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/input/button.html");
+            await Page.ClickAsync("button");
+            await Page.GoToAsync(TestConstants.ServerUrl + "/input/button.html");
+            await Page.ClickAsync("button");
+            Assert.Equal("Clicked", await Page.EvaluateExpressionAsync<string>("result"));
+        }
+
+        [Fact]
+        public async Task ShouldUploadTheFile()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/input/fileupload.html");
+            var filePath = "./assets/file-to-upload.txt";
+            var input = await Page.GetElementAsync("input");
+            await input.UploadFileAsync(filePath);
+            Assert.Equal("file-to-upload.txt", await Page.EvaluateFunctionAsync<string>("e => e.files[0].name", input));
+            Assert.Equal("contents of the file", await Page.EvaluateFunctionAsync(@"e => {
+                const reader = new FileReader();
+                const promise = new Promise(fulfill => reader.onload = fulfill);
+                reader.readAsText(e.files[0]);
+                return promise.then(() => reader.result);
+            }", input));
+        }
+
+        [Fact]
+        public async Task ShouldMoveWithTheArrowKeys()
+        {
+            await Page.GoToAsync(TestConstants.ServerUrl + "/input/textarea.html");
+            await Page.TypeAsync("textarea", "Hello World!");
+            Assert.Equal("Hello World!", await Page.EvaluateExpressionAsync<string>("document.querySelector('textarea').value"));
+            for (var i = 0; i < "World!".Length; i++)
+            {
+                Page.Keyboard.PressAsync("ArrowLeft");
```

Comment:
I'm getting several warnings because these guys are not being awaited. 
How about assigning the task to a dummy variable?

Reply:
Would it be okay to define a task field on the class and use it everywhere?

Reply:
Absolutely